plugins/scaffolder-field-extensions/scaffolder-frontend-module-http-request-field/src/components/SelectFieldFromApi/SelectFieldFromApi.tsx

```diff
@@ -110,12 +132,14 @@ export const SelectFieldFromApi = (props: FieldProps<string>) => {
 };
 
 export const selectFieldFromApiValidation = (
-  value: string,
+  value: unknown,
   validation: FieldValidation,
 ) => {
-  if (!KubernetesValidatorFunctions.isValidObjectName(value)) {
-    validation.addError(
-      'must start and end with an alphanumeric character, and contain only alphanumeric characters, hyphens, underscores, and periods. Maximum length is 63 characters.',
-    );
+  if (typeof value === 'string') {
+    if (!KubernetesValidatorFunctions.isValidObjectName(value)) {
+      validation.addError(
```

Comment:
Honestly.. I think this check is not correct. I dont see why every string returned from the field must be a valid kubernetes object name. It just seems wrong.

I think this validation should look at the schema type and make sure the javascripts types roughly match up.. e.g. a string type returns a string, a number returns a number, a boolean returns a boolean. etc.